### PR TITLE
fix: Empty `TextBlock` should have non-zero height

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/14201

## PR Type

What kind of change does this PR introduce?

- Bugfix

### <samp>🤖 Generated by Copilot at 6b96916</samp>

This pull request adds a new sample page for a `TextBox` with an animated header, and improves the line height calculation and rendering of the `TextBlock` control on Skia-based targets. It modifies several files in the `src` folder, such as `TextBlock.skia.cs`, `InlineCollection.skia.cs`, `Inline.skia.cs` and `FontDetails.skia.cs`, to use a new cached font line height value and simplify the layout logic. It also adds some new unit tests for the `TextBlock` control in the `Given_TextBlock.cs` file.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs [hunk 1]
```diff
@@ -43,7 +45,8 @@ protected override Size MeasureOverride(Size availableSize)
 		{
 			var padding = Padding;
 			var availableSizeWithoutPadding = availableSize.Subtract(padding);
-			var desiredSize = Inlines.Measure(availableSizeWithoutPadding);
+			var defaultLineHeight = (float)GetComputedLineHeight();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs:48]: I think the cast is not needed

```suggestion
			var defaultLineHeight = GetComputedLineHeight();
```

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs [hunk 2]
```diff
@@ -71,6 +74,35 @@ protected override Size ArrangeOverride(Size finalSize)
...
+		internal float GetComputedLineHeight()
+		{
+			var lineHeight = LineHeight;
+			if (!lineHeight.IsNaN() && lineHeight > 0)
+			{
+				return (float)lineHeight;
+			}
+			else
+			{
+				if (_cachedFontLineHeight is null)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs:92]: Do we really need to cache the the line height? I think the font itself is cached, then accessing `LineHeight` should be fast. Anything I'm missing?

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.skia.cs:92]: True!

**Review by ajpinedam — APPROVED** [review]
Great fix!

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs [hunk 3]
```diff
@@ -1,9 +1,6 @@
-﻿using System;
-using System.Linq;
+﻿using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Uno.UI.RuntimeTests.Helpers;
-using Uno.Helpers;
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBlock.cs:6]: Removing Uno.Helpers breaks Windows UWP build on CI. `WithOpacity` comes from here:

https://github.com/unoplatform/uno/blob/a8a16b80945c2ee4502c2bf42bd604eb932c63cc/src/Uno.UI.Toolkit/ColorExtensions.cs#L25-L28

**Comment by MartinZikmund** [comment]
@mergifyio backport release/stable/5.0

**Pull request merged**